### DISCUSSION TITLE
[mod integration] with Inventory Profiles Next

### DISCUSSION THREAD

**Discussion by blackd** [P3pp3rF1y/SophisticatedBackpacks#345] [Ideas]
Hi,

I'm developing [Inventory Profiles Next](https://www.curseforge.com/minecraft/mc-mods/inventory-profiles-next) mod. Lately I'm trying to extend support for modded storage containers. My mod generally works well with all kind of storages mods except for 2 things:
- overstacking - generally overstacking breaks how mouse clicks work
- detecting the dimensions as in high and width in slots 

To support these I need information about how big a stack can be, how many items will a single click on a stack take from the stack and of course the dimensions of the storage.

I did take a brief   look at your mod API but I wasn't able to find something that will allow me to get the above information.

Are you willing to make this kind of information easily accessible? The best case scenario will be if there is stable nice API that anyone implementing storage management can use. However can see why mod devs won't be so keen in implementing such API. 

I was thinking about something like few methods in net.p3pp3rf1y.sophisticatedbackpacks.common.gui.BackpackContainer with specific names or even better annotated with annotations that show what kind of information they provide.
.....

But of course the main question is are you interested in such integration at all :smile:?

**Comment by Halibear** [comment]
Just to comment... I am using both of your mods together now, haven't noticed anything weird (just put on Quark and that's a little funky) but no problems with sorting except it doesn't sort tools/weapons. (LOVE BOTH TOGETHER!!!)

**Comment by Konpaku-LVX** [comment]
Support Feature: Inventory Profiles Next: Slot Favouriting.

hey i realized this exists so im creating my thread here. @blackd its cool to see the developer of the mod here, i love this mod. @P3pp3rF1y i see this has gone unchecked for quite some time, hopefully a ping is alright in this case?

Inventory Profiles Next has a lock slot or favourite slot feature. basically, it will prevent items from being able to be moved. 

basically i want to lock my carrots and pickaxes inside my backpack, so i can use the shift + space feature to move everything BUT the items i absolutely want to stay in the backpack.

its an extremely useful feature when you need it

**Reply by blackd** [reply]
locking the slots of containers is not something I can easily do in IPN as it being client side.

@autoteelar I guess you should post separate feature request for @P3pp3rF1y to may be implement :)

**Reply by Konpaku-LVX** [reply]
makes sense, thank you! also if anyone looking for a workaround, you can do slot memory in sophisticated backpacks, it kind of locks the slots a bit :D
